app/helpers/zones_helper.rb

```diff
@@ -0,0 +1,30 @@
+module ZonesHelper
+  private
+
+  def settings_zone_list(zones, my_zone_name)
+    {
+      :headers  => [{:header => _("Zones"), :key => 'zone_type'}],
+      :rows     => settings_zone_rows(zones),
+      :selected => selected_zone(zones, my_zone_name)
+    }
+  end
+
+  def selected_zone(zones, my_zone_name)
+    !zones.empty? ? zones.where('name = ?', my_zone_name)&.first&.id&.to_s : []
```

Comment:
```suggestion
    !zones.empty? ? zones.where(:name => my_zone_name).first&.id&.to_s : []
```

Reply:
Actually I don't understand this method at all.  If zones is an Array, we can just use `.detect`, but then I noticed the blank case returns an Array, and the non-blank case returns a single String (which is the id number).

If we were using .detect, I'd expect something like:

```suggestion
    zones.detect { |z| z.name == my_zone_name }&.id&.to_s
```